invokeai/app/api/routers/models.py

```diff
@@ -54,226 +66,70 @@ async def list_models() -> ModelsList:
-        #             model_merge_info["force"],
-        #         )
-
-        #         dump_path = global_models_dir() / "merged_models"
-        #         if model_merge_info["model_merge_save_path"] is not None:
-        #             dump_path = Path(model_merge_info["model_merge_save_path"])
-
-        #         os.makedirs(dump_path, exist_ok=True)
-        #         dump_path = dump_path / model_merge_info["merged_model_name"]
-        #         merged_pipe.save_pretrained(dump_path, safe_serialization=1)
-
-        #         merged_model_config = dict(
-        #             model_name=model_merge_info["merged_model_name"],
-        #             description=f'Merge of models {", ".join(models_to_merge)}',
-        #             commit_to_conf=opt.conf,
-        #         )
-
-        #         if vae := self.generate.model_manager.config[models_to_merge[0]].get(
-        #             "vae", None
-        #         ):
-        #             print(f">> Using configured VAE assigned to {models_to_merge[0]}")
-        #             merged_model_config.update(vae=vae)
-
-        #         self.generate.model_manager.import_diffuser_model(
-        #             dump_path, **merged_model_config
-        #         )
-        #         new_model_list = self.generate.model_manager.list_models()
-
-        #         socketio.emit(
-        #             "modelsMerged",
-        #             {
-        #                 "merged_models": models_to_merge,
-        #                 "merged_model_name": model_merge_info["merged_model_name"],
-        #                 "model_list": new_model_list,
-        #                 "update": True,
-        #             },
-        #         )
-        #         print(f">> Models Merged: {models_to_merge}")
-        #         print(f">> New Model Added: {model_merge_info['merged_model_name']}")
-        #     except Exception as e:
-        #         self.handle_exceptions(e)
\ No newline at end of file
+@models_router.post(
+    "/",
+    operation_id="update_model",
+    responses={
+        201: {
+        "model_response": "Model added", 
+        },
+        202: {
+        "description": "Model submission is processing. Check back later."
+        }, 
+    },
+)
+async def update_model(
+    model_request: CreateModelRequest
+) -> CreateModelResponse:
+    """ Add Model """
+    try:
+        model_request_info = model_request.info
+        print(f">> Checking for {model_request_info}...")
+        info_dict = model_request_info.dict()
+
+        ApiDependencies.invoker.services.model_manager.add_model(
+            model_name=model_request.name,
+            model_attributes=info_dict,
+            clobber=True,
+        )
+        model_response = CreateModelResponse(name=model_request.name, info=model_request.info, status="success")
+
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+    
+    return model_response
```

Comment:
I don't think this will return 201 naturally. A 201 has a `Location` header that points to the created/updated resource. I think returning a model just returns 200.

Similar with 202 - I see it documented, but that's never returned here.

You might need to wrap something around the add_model that will wait a second or two, then time out and return a 202 if it hasn't completed yet (but still leave it running on the server).

Reply:
I've taken away the 202 reference entirely. Timing on this doesn't justify the timeout - it's near instant. And I think anything that would cause it to timeout probably will significantly reduce likelihood of success.